alerting/src/test/kotlin/org/opensearch/alerting/bwc/AlertingBackwardsCompatibilityIT.kt

```diff
@@ -59,18 +59,17 @@ class AlertingBackwardsCompatibilityIT : AlertingRestTestCase() {
                 ClusterType.MIXED -> {
                     assertTrue(pluginNames.contains("opensearch-alerting"))
                     verifyMonitorExists(LEGACY_OPENDISTRO_ALERTING_BASE_URI)
-                    // Waiting a minute to ensure the Monitor ran again at least once before checking if the job is running
-                    // on time
-                    // TODO: Should probably change the next execution time of the Monitor manually instead since this inflates
-                    //  the test execution by a lot
-                    Thread.sleep(60000)
                     // TODO: Need to move the base URI being used here into a constant and rename ALERTING_BASE_URI to
                     //  MONITOR_BASE_URI
                     verifyMonitorStats("/_opendistro/_alerting")
                 }
                 ClusterType.UPGRADED -> {
                     assertTrue(pluginNames.contains("opensearch-alerting"))
                     verifyMonitorExists(ALERTING_BASE_URI)
+                    // TODO: Change the next execution time of the Monitor manually instead since this inflates
+                    //  the test execution by a lot (might have to wait for Job Scheduler plugin integration first)
+                    // Waiting a minute to ensure the Monitor ran again at least once before checking if the job is running
+                    // on time
                     Thread.sleep(60000)
```

Comment:
Usually a consistent way to avoid sleeps is by using ``assertBusy``, given if we expect some change of state within a stipulated time threshold. It allows to runs the code block for the provided interval, waiting for no assertions to trip, and return as soon as all the assertions are met. So wondering of there exists some state related to monitor execution, such as count, that we can rely upon as a trigger here, and then verify the stats response.

Reply:
The reason for the sleep was that the minimum Monitor execution interval is `1 minute` and we were going to ensure the Monitor executed again before checking the stats to make sure that any late running Monitor manifested itself. The test itself is not perfect and we'd normally want to just fast forward the job execution like we do for the Index Management test but Alerting uses an older job scheduling implementation where the next job execution time is not stored on the document but rather in some `ConcurrentHashMap` object that is not test friendly.

Reply:
Thanks for clarifying @qreshi. So my suggestion was, is there is a way (API/Object) which can specify if monitor was run again (such as a running-counter). If yes, then we can do [assertBusy](https://github.com/opensearch-project/OpenSearch/blob/main/test/framework/src/main/java/org/opensearch/test/OpenSearchTestCase.java#L1053-L1055), which provides an exponential backoff mechanism to wait until the condition is met.  
If not, I understand having sleep is the only valid option.